src/java/main/org/apache/zookeeper/server/quorum/Learner.java

```diff
@@ -84,8 +84,12 @@ public Socket getSocket() {
     
     protected static final Logger LOG = LoggerFactory.getLogger(Learner.class);
 
+    private static final int leaderConnectDelayDuringRetryMs =
+            Integer.getInteger("zookeeper.leaderConnectDelayDuringRetryMs", 100);
```

Comment:
The change was inspired by an Observer potentially taking a long time to recover after a leader election if the first LearnerMaster that it hooked in to was a Follower that was lagging in syncing with the leader. Rather than fixate on this slow Follower, the Observer can be moved along to a new LearnerMaster with minimal delay.